modules/reactor/src/meshgenerators/PatternedCartesianMeshGenerator.C

```diff
@@ -0,0 +1,744 @@
+      else
+        _pattern_pitch = getParam<Real>("square_size");
+    }
+    for (MooseIndex(_input_names) i = 0; i < _input_names.size(); ++i)
+    {
+      // throw an error message if the input mesh does not contain the required meta data
+      if (!hasMeshProperty("pitch_meta", _input_names[i]))
+        mooseError("In PatternedCartesianMeshGenerator ",
+                   _name,
+                   ": the unit square input mesh does not contain appropriate meta data "
+                   "required for generating an assembly.");
+      pitch_array.push_back(getMeshProperty<Real>("pitch_meta", _input_names[i]));
+
+      num_sectors_per_side_array_tmp =
+          getMeshProperty<std::vector<unsigned int>>("num_sectors_per_side_meta", _input_names[i]);
+      if (*std::max_element(num_sectors_per_side_array_tmp.begin(),
+                            num_sectors_per_side_array_tmp.end()) !=
+          *std::min_element(num_sectors_per_side_array_tmp.begin(),
+                            num_sectors_per_side_array_tmp.end()))
+        mooseError("In PatternedCartesianMeshGenerator ",
+                   _name,
+                   ": num_sectors_per_side metadata values of all four sides of each input mesh "
+                   "generator must be identical.");
+      num_sectors_per_side_array.push_back(*num_sectors_per_side_array_tmp.begin());
+      background_intervals_array.push_back(
+          getMeshProperty<unsigned int>("background_intervals_meta", _input_names[i]));
+      node_id_background_array.push_back(
+          getMeshProperty<dof_id_type>("node_id_background_meta", _input_names[i]));
+      max_radius_array.push_back(getMeshProperty<Real>("max_radius_meta", _input_names[i]));
+    }
+    max_radius_global = *max_element(max_radius_array.begin(), max_radius_array.end());
+    if (!MooseUtils::absoluteFuzzyEqual(*std::max_element(pitch_array.begin(), pitch_array.end()),
+                                        *std::min_element(pitch_array.begin(), pitch_array.end())))
+      mooseError("In PatternedCartesianMeshGenerator ",
+                 _name,
+                 ": pitch metadata values of all input mesh generators must be identical.");
+    _input_pitch_meta = pitch_array.front();
+    if (*std::max_element(num_sectors_per_side_array.begin(), num_sectors_per_side_array.end()) !=
+        *std::min_element(num_sectors_per_side_array.begin(), num_sectors_per_side_array.end()))
+      mooseError(
+          "In PatternedCartesianMeshGenerator ",
+          _name,
+          ": num_sectors_per_side metadata values of all input mesh generators must be identical.");
+  }
+
+  std::vector<Real> extra_dist;
+  Real extra_dist_shift(0.0);
+  Real y_min(0.0);
+  Real y_max_0(0.0);
+  Real y_max_n(0.0);
+  const Real extra_dist_tol = _pattern_boundary == "expanded" ? pitch_array.front() / 10.0 : 0.0;
+  const Real extra_dist_shift_0 = _pattern_boundary == "expanded" ? pitch_array.front() / 5.0 : 0.0;
```

Comment:
why the 5 here?

pitch_array is not necessarily symmetric between X and Y either right?

Reply:
It's an arbitrary number...

Reply:
As this is actually square geometry, it is symmetric.